### PR TITLE
PLAT-62 adding a resource picker to the block section

### PR DESCRIPTION
**Pull request by lefterisk**
For the week and month view that dont have a pre selected resource

### CODE REVIEW

**Comment by Glenn** [comment]
Can we call people staff or people, and not "resources"

Also - can we extend this also also support actual resources (and detect if they aren't configured)

**Comment by lefterisk** [comment]
Added the ability to select resource as well as a person. Both lists are merged into the same select which has OptionGroups 'Staff' and 'Resources'

File: src/admin-booking/templates/quick_pick.html [hunk 1]
```diff
@@ -25,15 +25,14 @@
...
     <h1>Block Time</h1>
-
     <div class="form-group">
       <label>For</label>
-      <p ng-show="bb.current_item.person.name">
-        {{bb.current_item.person.name}}
-      </p>
       <p ng-show="bb.current_item.resource.name">
         {{bb.current_item.resource.name}}
       </p>
```
Comment (lukeferguson) [inline src/admin-booking/templates/quick_pick.html]: this can be deleted too

Comment (lefterisk) [inline src/admin-booking/templates/quick_pick.html]: Done

File: src/admin-booking/javascripts/directives/block-time.js.coffee [hunk 2]
```diff
@@ -2,6 +2,47 @@ angular.module('BBAdminBooking').directive 'bbBlockTime', () ->
...
+    # If in "Day" view a person or resource will have been passed in
+    if $scope.bb.current_item.person? and $scope.bb.current_item.person.id?
+      $scope.picked_resource = $scope.bb.current_item.person.id + '_p'
+
+    if $scope.bb.current_item.resource? and $scope.bb.current_item.resource.id?
+      $scope.picked_resource = $scope.bb.current_item.resource.id  + '_r' 
+
+    # On select change update the right current_item variable depending 
+    # whether the selected item is a person or a resource
+    $scope.$watch 'picked_resource', (newValue, oldValue)->
```
Comment (lukeferguson) [inline src/admin-booking/javascripts/directives/block-time.js.coffee]: use of ng-change bound to the select control would be appropriate - $watch gets evaluated every digest cycle whereas ng-change is only invoked when the drop down changes

Comment (lefterisk) [inline src/admin-booking/javascripts/directives/block-time.js.coffee]: Swapped $watch with ng-change

**Pull request merged**